### DISCUSSION TITLE
Using NuxtLink in Nuxt 3 component library got broken after recent update

### DISCUSSION THREAD

**Discussion by vitpetricak** [nuxt/nuxt#16474] [Questions]
Hello, I have a problem with using NuxtLink in component library.

My setup is:
- custom component library in separate package
- 3 apps using the library to keep the feel consistent

In the library I have wrappers for sidebar links / nested sidebar / breadcrumbs etc, using NuxtLink (mocked in Storybook as basic links, later replaced in apps by the real component)

For example my breadcrumbs:
```
<div class="flex gap-5 items-center">
  <template v-for="(item, i) in items" :key="item.to">
    <div class="flex items-center">
      <NuxtLink
        :to="item.to"
        :class="(i === (items.length - 1) || 'underline')">
        {{ item.label }}
      </NuxtLink>
      <div v-if="!(i === (items.length - 1))" class="ml-5 w-2 text-gray-250">
        <DownArrowIcon class="transform -rotate-90" />
      </div>
    </div>
  </template>
</div>
```

Until recently it was working fine, but after latest update, the replacing of component completely broke.

Now I am getting error
```
runtime-core.esm-bundler.js:38 [Vue warn]: Failed to resolve component: nuxt-link
If this is a native custom element, make sure to exclude it from component resolution via compilerOptions.isCustomElement.
```

and component is not replaced. Any ideas how to go around this?

I tried
- changing the name of component to NuxtLink, nuxt-link, nuxtlink
- putting it into the vue config `compilerOptions.isCustomElement` in the nuxt project
- putting it into the vue config `compilerOptions.isCustomElement` in the library

**Comment by vitpetricak** [comment]
As of now, my solution is:
- Renaming all NuxtLink to UiLink
- Add component UiLink in plugin with defineNuxtLink
```
app.vueApp.component('UiLink', defineNuxtLink({
    componentName: 'UiLink'
}))
```
However I find this solution not ideal, so if anyone has good idea, I'll be happy to try it.

**Comment by tkyisgk** [comment]
I have a same problem.

**Comment by tjifyodor** [comment]
Using `router-link` instead is a possible fix.

**Comment by wsaca** [comment]
It's a problem, resolveComponent() resolves router-link but not nuxt-link.

**Reply by TheAlexLichter** [reply]
Please raise an issue with reproduction then ☺️
